### PR TITLE
stream_info: moving downstream timing into its own class

### PR DESCRIPTION
**Pull request by alyssawilk**
Companion PR to https://github.com/envoyproxy/envoy/pull/18934

Risk Level: low (refactor)
Testing: n/a
Docs Changes: n/a
Release Notes: n/a

Part of envoyproxy/envoy-mobile#1520

### CODE REVIEW

File: envoy/stream_info/stream_info.h [hunk 1]
```diff
@@ -238,6 +238,41 @@ struct UpstreamTiming {
...
+    ASSERT(!first_downstream_tx_byte_sent_);
+    first_downstream_tx_byte_sent_ = time_source.monotonicTime();
+  }
+  void onLastDownstreamTxByteSent(TimeSource& time_source) {
+    ASSERT(!last_downstream_tx_byte_sent_);
+    last_downstream_tx_byte_sent_ = time_source.monotonicTime();
+  }
+
+private:
+  absl::flat_hash_map<std::string, MonotonicTime> timings_;
```
Comment (zuercher) [inline envoy/stream_info/stream_info.h:267]: I don't see where this is used. Is it for a follow-on change?

**Review by zuercher — APPROVED** [review]
Looks good. Just had one question.

File: envoy/stream_info/stream_info.h (hunk 1 above)
Comment (alyssawilk) [reply envoy/stream_info/stream_info.h:267]: indeed, sorry I failed to clean up here, it's part of https://github.com/envoyproxy/envoy/pull/18934

**Pull request merged**